model/src/main/proto/profile.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package model;
+
+option java_package = "org.oppia.app.model";
+option java_multiple_files = true;
+
+// Top level proto used to store last used id and all profiles
+message Profiles {
+  // Represents the last used unique ID for adding a profile
+  int32 last_used_id = 1;
+
+  // Mapping from unique ID to profile
+  map<int32, Profile> profiles = 2;
+}
+
+// Structure for a single profile
+message Profile {
+  // Unique ID given to each profile
+  int32 id = 1;
+
+  // Name of the user
+  string name = 2;
+
+  // Determines whether profile uses PIN
+  bool hasPin = 3;
+
+  // PIN of the profile
+  string pin = 4;
+
+  // User selected image or gravatar
```

Comment:
How do we tell the difference? Or does it not matter since they're both represented by URIs?

Also, maybe say ``avatar_image_uri`` to indicate that content URIs (https://developer.android.com/guide/topics/providers/content-provider-basics#ContentURIs) are valid here in addition to remote URLs.